src/zino/tasks/bfdtask.py

```diff
@@ -0,0 +1,114 @@
+        ("BFD-STD-MIB", "bfdSessAddr"),
+        ("BFD-STD-MIB", "bfdSessAddrType"),
+    ]
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self._scheduler = get_scheduler()
+        self._snmp = SNMP(self.device)
+
+    async def run(self):
+        if self._device_state.is_juniper:
+            polled_state = await self._poll_juniper()
+            self._update_state_for_all_ports(polled_state)
+
+    def _update_state_for_all_ports(self, polled_state: BFDStates):
+        for port in self._device_state.ports.values():
+            new_state = polled_state.get(port.ifdescr, BFDState(session_state=BFDSessState.NO_SESSION))
+            self._update_state(port, new_state)
+
+    def _update_state(self, port: Port, new_state: BFDState):
+        """Updates the BFD state for a port. Will create or update BFD events depending on the state changes"""
+        # Do not create event if this is the first time BFD state is polled for this port
+        if port.bfd_state:
+            if port.bfd_state.session_state != new_state.session_state:
+                self._create_or_update_event(port, new_state)
+        port.bfd_state = new_state
+
+    def _create_or_update_event(self, port: Port, new_state: BFDState):
+        event, created = self.state.events.get_or_create_event(self.device.name, port.ifindex, BFDEvent)
+        if created:
+            event.state = EventState.OPEN
+            event.add_history("Change state to Open")
+
+        event.bfdstate = new_state.session_state
+        event.bfdix = new_state.session_index
+        event.bfddiscr = new_state.session_discr
+        event.bfdaddr = new_state.session_addr
+
+        log = f"Port {port.ifdescr} changed BFD state from {port.bfd_state.session_state} to {new_state.session_state}"
+        event.add_log(log)
+
+    async def _poll_juniper(self) -> BFDStates:
+        bfd_rows = await self._snmp.sparsewalk(*self.JUNIPER_BFD_COLUMNS)
+        bfd_states = self._parse_juniper_rows(bfd_rows)
+        return bfd_states
+
+    def _parse_juniper_rows(self, bfd_rows: SparseWalkResponse) -> BFDStates:
+        """The keys for the return dict should match the respective interface's IfDescr value"""
+        bfd_states = {}
+        for index, row in bfd_rows.items():
+            interface_name = row["jnxBfdSessIntfName"]
+            session_state = row["bfdSessState"]
+            session_discr = row["bfdSessDiscriminator"]
+            session_addr = row["bfdSessAddr"]  # This is a string containing bytes (ex. '\x7f\x00\00x\01')
+            session_addr_type = row["bfdSessAddrType"]
+
+            try:
+                encoded_bytes = str.encode(session_addr, "utf-8")
+                ipaddr = self._convert_address(encoded_bytes, session_addr_type)
+            except ValueError as e:
+                _log.error(f"Error converting bfdSessAddr object to an IP address: {e}")
+                ipaddr = None
```

Comment:
This error log lacks one important piece of information: Which device had the "bad" response?

Testing this branch, I get reams of this error - is that your experience as well?

```
ERROR - zino.tasks.bfdtask (MainThread) - Error converting bfdSessAddr object to an IP address: b"REDACTED" (len 7 != 4) is not permitted as an IPv4 address
ERROR - zino.tasks.bfdtask (MainThread) - Error converting bfdSessAddr object to an IP address: b"REDACTED" (len 6 != 4) is not permitted as an IPv4 address
ERROR - zino.tasks.bfdtask (MainThread) - Error converting bfdSessAddr object to an IP address: b"REDACTED" (len 7 != 4) is not permitted as an IPv4 address
ERROR - zino.tasks.bfdtask (MainThread) - Error converting bfdSessAddr object to an IP address: b"REDACTED" (len 6 != 4) is not permitted as an IPv4 address
ERROR - zino.tasks.bfdtask (MainThread) - Error converting bfdSessAddr object to an IP address: b'REDACTED' (len 5 != 4) is not permitted as an IPv4 address
ERROR - zino.tasks.bfdtask (MainThread) - Error converting bfdSessAddr object to an IP address: b'REDACTED' (len 5 != 4) is not permitted as an IPv4 address
```

I've redacted the actual data, stop by my desk if you need it :)

Reply:
I can swear this used to work.. Sometimes pysnmp wants to give values that makes sense, sometimes it does not.. 
These errors are caused by for example the hexstring  `80 27 FE 69` gets represented as   `\xc2\x80'\xc3\xbei`, which makes little sense to me.

Changing the parsing of octetstrings to using prettyPrint() instead of just using str() seems to fix it without breaking other stuff. I used prettyPrint() previously, but managed to get it to work without making that change (or so I thought), but it seems to be necessary after all.